iOSDevUK/Controllers/Sessions/ProgrammeTableViewController.swift

```diff
@@ -86,6 +98,41 @@ class ProgrammeTableViewController: UIViewController, UITableViewDelegate, UITab
         }
     }
     
+    /// Find the current or next session for the current set of sessions and scroll to it.
+    func scrollToCurrentSession(animated: Bool) {
+        guard let fetchedResultsController = fetchedResultsController,
+                let sections = fetchedResultsController.sections else {
+            return
+        }
+        
+        /// Find the session in the already-fetched results
+        let now = Date()
+        let nextOrCurrentSession = fetchedResultsController.fetchedObjects?.first { session in
+            guard let startTimeNSDate = session.startTime,
+                    let endTimeNSDate = session.endTime else {
+                return false
+            }
+            let startTime = startTimeNSDate as Date
+            let endTime = endTimeNSDate as Date
+            return (startTime <= now && endTime >= now) || (startTime >= now)
```

Comment:
I think we can assume that `startTime >= endTime will always be true. And in that case, we can simplify it as 
```swift
return return endTime > now
```
which basically says "give me the first session for which the endTime is in the future".

The conversion from `NSDate` (which is what the CoreData property is) to `Date` is a bit cumbersome, but I get that you do it to be able to use the nicer comparison with `<=`. 

If you want to avoid the conversion and still have reasonably nice code, you could do
```swift
return endTimeNSDate.timeIntervalSinceNow > 0
```

Overall, we could then get away with 
```swift
    /// Find the session in the already-fetched results
    let nextOrCurrentSession = fetchedResultsController.fetchedObjects?.first { session in
        guard let endTime = session.endTime else { return false }
        return endTime.timeIntervalSinceNow > 0
    }
```

Reply:
Yes, a good simplification. I was trying to handle all possible poor data scenarios of course :)